src/lib/structures/symbols.ml

```diff
@@ -317,82 +296,209 @@ let name_to_string =
+    | Op op -> pp_operator ppf op
+
+    (* Core theory *)
+    | True -> Fmt.pf ppf "true"
+    | False -> Fmt.pf ppf "false"
+    | Void -> Fmt.pf ppf "void"
+    | Name (n, _, _) -> pp_name ppf (Hstring.view n)
+    | Var v when show_vars -> Fmt.pf ppf "'%s'" (Var.to_string v)
+    | Var v -> Fmt.string ppf (Var.to_string v)
+
+    (* Reals and Ints theories *)
+    | Int i -> Z.pp_print ppf i
+    | Real q -> Q.pp_print ppf q
+
+    (* FixedSizedBitVectors theory *)
+    | Bitv (n, s) ->
+      Fmt.pf ppf "[|%s|]" (Z.format (Fmt.str "%%0%db" n) s)
+
+    (* Symbols used in semantic triggers *)
+    | In (lb, rb) ->
+      Fmt.pf ppf "%s, %s" (string_of_bound lb) (string_of_bound rb)
+    | MapsTo v -> Fmt.pf ppf "%a |->" Var.print v
+end
+
+module SmtPrinter = struct
+  let pp_operator ppf op =
+    match op with
+    (* Core theory *)
+    |Tite -> Fmt.pf ppf "ite"
+
+    (* Reals and Ints theories *)
+    | Plus -> Fmt.pf ppf "+"
+    | Minus -> Fmt.pf ppf "-"
+    | Mult -> Fmt.pf ppf "*"
+    | Div -> Fmt.pf ppf "/"
+    | Modulo -> Fmt.pf ppf "%%"
+
+    (* Reals_Ints theory *)
+    | Abs_int | Abs_real -> Fmt.pf ppf "abs"
+    | Real_of_int -> Fmt.pf ppf "to_real"
+    | Real_is_int -> Fmt.pf ppf "is_int"
+    | Integer_round -> Fmt.pf ppf "to_int"
+
+    (* FixedSizedBitVectors theory *)
+    | Extract (i, j) ->
+      Fmt.pf ppf "(_ extract %d %d)" i j
+    | Concat -> Fmt.pf ppf "concat"
+    | BV2Nat -> Fmt.pf ppf "bv2nat"
+    | BVnot -> Fmt.pf ppf "bvnot"
+    | BVand -> Fmt.pf ppf "bvand"
+    | BVor -> Fmt.pf ppf "bvor"
+
+    (* ArraysEx theory *)
+    | Get -> Fmt.pf ppf "select"
+    | Set -> Fmt.pf ppf "store"
+
+    (* DT theory *)
+    | Record -> ()
+    | Access s | Constr s | Destruct (s, _) -> Hstring.print ppf s
+
+    (* Float theory *)
+    | Float -> Fmt.pf ppf "ae.round"
+
+    (* Not in the SMT-LIB standard *)
+    | Int2BV _ -> Fmt.pf ppf "(_ int2bv)"
+    | Not_theory_constant -> Fmt.pf ppf "ae.not_theory_constant"
+    | Is_theory_constant -> Fmt.pf ppf "ae.is_theory_constant"
+    | Linear_dependency -> Fmt.pf ppf "ae.linear_dependency"
+    | Sqrt_real -> Fmt.pf ppf "ae.sqrt_real"
+    | Sqrt_real_default -> Fmt.pf ppf "ae.sqrt_real_default"
+    | Sqrt_real_excess -> Fmt.pf ppf "ae.sqrt_real_excess"
+    | Int_floor -> Fmt.pf ppf "ae.int_floor"
+    | Int_ceil -> Fmt.pf ppf "ae.int_ceil"
+    | Max_real -> Fmt.pf ppf "ae.max_real"
+    | Max_int -> Fmt.pf ppf "ae.max_int"
+    | Min_real -> Fmt.pf ppf "ae.min_real"
+    | Min_int -> Fmt.pf ppf "ae.min_int"
+    | Integer_log2 -> Fmt.pf ppf "ae.integer_log2"
+    | Pow -> Fmt.pf ppf "ae.pow"
+
+    | Optimize _ ->
+      (* TODO: this case will be removed in the PR #921. *)
+      assert false
+
+    | Fixed | Reach ->
+      (* Unused symbols *)
+      assert false
+
+end
+
+let[@inline always] pp_operator ~(format:[`Ae | `Smtlib]) ppf op =
+  match format with
+  | `Ae -> AEPrinter.pp_operator ppf op
+  | `Smtlib -> SmtPrinter.pp_operator ppf op
```

Comment:
I am not sure the polymorphic variants are very useful here, ``pp_operator ~format:`Ae`` and ``pp_operator ~format:`Smtlib`` are harder to type than separate names such as `pp_ae_operator` and `pp_smt_operator` which also wouldn't require additional parentheses when used with `Fmt.pf`.

Or really we could just expose the `SmtPrinter` and `AEPrinter` modules directly I think.

Reply:
For me both solutions are equivalent.